x-pack/plugins/secops/public/components/timeline/body/renderers/suricata_signature.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiBadge, EuiFlexGroup, EuiFlexItem, EuiLink } from '@elastic/eui';
+import * as React from 'react';
+import { pure } from 'recompose';
+import styled from 'styled-components';
+
+import { DragEffects, DraggableWrapper } from '../../../drag_and_drop/draggable_wrapper';
+import { escapeDataProviderId } from '../../../drag_and_drop/helpers';
+import { Provider } from '../../../timeline/data_providers/provider';
+import { getBeginningTokens } from './suricata_links';
+
+const SignatureFlexItem = styled(EuiFlexItem)`
+  min-width: 77px;
+`;
+
+const Badge = styled(EuiBadge)`
+  vertical-align: top;
+`;
+
+const TokensFlexItem = styled(EuiFlexItem)`
+  margin-left: 3px;
+`;
+
+export const GoogleLink = pure(({ link, value }: { link: string; value: string }) => (
+  <TokensFlexItem grow={false}>
+    <EuiLink href={`https://www.google.com/search?q=${encodeURI(link)}`} target="_blank">
+      {value}
+    </EuiLink>
+  </TokensFlexItem>
+));
+
+export const Tokens = pure(({ tokens }: { tokens: string[] }) => (
+  <>
+    {tokens.map(token => (
+      <TokensFlexItem key={token} grow={false}>
+        <EuiBadge iconType="tag" color="hollow">
+          {token}
+        </EuiBadge>
+      </TokensFlexItem>
+    ))}
+  </>
+));
+
+export const DraggableSignatureId = pure(
+  ({ id, signatureId }: { id: string; signatureId: string }) => (
+    <SignatureFlexItem grow={false}>
+      <DraggableWrapper
+        dataProvider={{
+          and: [],
+          enabled: true,
+          id: escapeDataProviderId(`suricata-${id}-sig-${signatureId}`),
+          name: signatureId,
+          excluded: false,
+          kqlQuery: '',
+          queryMatch: {
+            field: 'suricata.eve.alert.signature_id',
+            value: signatureId,
+          },
+        }}
+        render={(dataProvider, _, snapshot) =>
+          snapshot.isDragging ? (
+            <DragEffects>
+              <Provider dataProvider={dataProvider} />
+            </DragEffects>
+          ) : (
+            <Badge iconType="number" color="hollow">
+              {signatureId}
+            </Badge>
+          )
+        }
+      />
+    </SignatureFlexItem>
+  )
+);
+
+export const SuricataSignature = pure(
+  ({ id, signature, signatureId }: { id: string; signature: string; signatureId: string }) => {
+    const tokens = getBeginningTokens(signature);
+    return (
+      <EuiFlexGroup justifyContent="center" gutterSize="none">
+        <DraggableSignatureId id={id} signatureId={signatureId} />
+        <Tokens tokens={tokens} />
+        <GoogleLink
```

Comment:
UI Nit: Perhaps add additional margin between the last token and the `GoogleLink` so that when the link is in the `:focus` state it isn't touching the previous token.

![image](https://user-images.githubusercontent.com/2946766/53456617-4f7a7000-39ec-11e9-8c25-0dfb72852f2f.png)

Reply:
Good eye. I added a bit more for just the link and now it looks like this below. Much better.

<img width="592" alt="screen shot 2019-02-27 at 9 13 40 pm" src="https://user-images.githubusercontent.com/1151048/53540881-95aafe80-3ad4-11e9-904a-216caaf2cf74.png">